src/lnp/application/payment/types.rs

```diff
@@ -256,3 +261,134 @@ impl DumbDefault for TempChannelId {
         Self(Default::default())
     }
 }
+
+#[derive(Wrapper, Clone, Debug, From, PartialEq, Eq)]
+pub struct NodeColor([u8; 3]);
+
+impl LightningEncode for NodeColor {
+    fn lightning_encode<E: io::Write>(
+        &self,
+        mut e: E,
+    ) -> Result<usize, io::Error> {
+        let len = e.write(self.as_inner())?;
+        Ok(len)
+    }
+}
+
+impl LightningDecode for NodeColor {
+    fn lightning_decode<D: io::Read>(mut d: D) -> Result<Self, LightningError> {
+        let mut buf = [0u8; 3];
+        d.read_exact(&mut buf)?;
+        Ok(Self::from_inner(buf))
+    }
+}
+
+#[derive(Wrapper, Clone, Debug, From, PartialEq, Eq)]
+pub struct Alias([u8; 3]);
+
+impl LightningEncode for Alias {
+    fn lightning_encode<E: io::Write>(
+        &self,
+        mut e: E,
+    ) -> Result<usize, io::Error> {
+        let len = e.write(self.as_inner())?;
+        Ok(len)
+    }
+}
+
+impl LightningDecode for Alias {
+    fn lightning_decode<D: io::Read>(mut d: D) -> Result<Self, LightningError> {
+        let mut buf = [0u8; 3];
+        d.read_exact(&mut buf)?;
+        Ok(Self::from_inner(buf))
+    }
+}
+
+/// Lightning network short channel Id as per BIP7
+#[derive(
+    Clone,
+    Copy,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+    Debug,
+    Default,
+    From,
+    StrictEncode,
+    StrictDecode,
+)]
+#[lnpbp_crate(crate)]
+pub struct ShortChannelId {
```

Comment:
We need to provide a custom Display here according to BOLT-7 https://github.com/lightningnetwork/lightning-rfc/blob/master/07-routing-gossip.md#definition-of-short_channel_id

```text
The standard human readable format for short_channel_id is created by printing the above components, in the order: block height, transaction index, and output index. Each component is printed as a decimal number, and separated from each other by the small letter x. For example, a short_channel_id might be written as 539268x845x1, indicating a channel on the output 1 of the transaction at index 845 of the block at height 539268.
```

Reply:
Done